clang/lib/DPCT/Utility.h

```diff
@@ -581,6 +581,19 @@ class PairedPrinter {
   ~PairedPrinter() { OS << Postfix; }
 };
 std::string appendPath(const std::string &P1, const std::string &P2);
+std::set<const clang::DeclRefExpr *>
+matchTargetDREInScope(const clang::VarDecl *TargetDecl,
+                      const clang::Stmt *Range);
+/// @brief Check if an argument is initialized.
+/// @param Arg Function call argument.
+/// @param DeclsNeedToBeInitialized UnInitialized VarDecl(s). Only can be used
+/// when return 0.
+/// @return  1: Initialized
+///          0: Not initialized.
+///         -1: Cannot deduce.
+int isArgumentInitialized(
+    const clang::Expr *Arg,
+    std::vector<const clang::VarDecl *> &DeclsNeedToBeInitialized);
```

Comment:
Does this function work for pointer type or array type? If not, should we need to consider above two type?
Like:
__shfl_xor_sync(0xFFFFFFFF, val[index] or *val, laneMask, WS);

Reply:
This issue only occurs for scalar type parameters.